### PR TITLE
Fix to make it work on WSL (Windows Service for Linux)

### PR DESCRIPTION
**Pull request by marcosdiez**
On WSL (Windows 10 v1803 (17134.1)), some non relevant interfaces do not support `self._listen_socket.setsockopt(
socket.IPPROTO_IP, socket.IP_ADD_MEMBERSHIP,
socket.IPPROTO_IP, socket.IP_ADD_MEMBERSHIP, _value)`

This line make we gently ignore this error and continue to work properly.

Here is the output before:

```
(env) mdiez@Tiburaska:~/m/catt/x/python-zeroconf$ python self_test.py
Multicast DNS Service Discovery for Python, version 0.20.0
Traceback (most recent call last):
  File "self_test.py", line 18, in <module>
    r = Zeroconf()
  File "/mnt/c/marcos/linux/catt/x/python-zeroconf/zeroconf.py", line 1710, in __init__
    socket.IPPROTO_IP, socket.IP_ADD_MEMBERSHIP, _value)
OSError: [Errno 22] Invalid argument
```
and after my fix:

```
(env) mdiez@Tiburaska:~/m/catt/x/python-zeroconf$ python self_test.py
Multicast DNS Service Discovery for Python, version 0.20.0
1. Testing registration of a service...
   Registering service...
   Registration done.
2. Testing query of service information...
   Getting ZOE service: None
   Query done.
3. Testing query of own service...
   Getting self: ServiceInfo(type='_http._tcp.local.', name='My Service Name._http._tcp.local.', address=b'\x7f\x00\x00\x01', port=1234, weight=0, priority=0, server='My Service Name._http._tcp.local.', properties={b'version': b'0.10', b'a': b'test value', b'b': b'another value'})
   Query done.
4. Testing unregister of service information...
   Unregister done.
(env) mdiez@Tiburaska:~/m/catt/x/python-zeroconf$

```

Please look only at the first commit for the code review. The rest is just `flake8` and `travis`

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -10,6 +10,7 @@ install:
     - pip install -r requirements-dev.txt
 script:
     - make test_coverage
+    - flake8 --version
```
Comment (jstasiak) [inline .travis.yml]: This doesn't seem necessary?

File: requirements-dev.txt [hunk 2]
```diff
@@ -10,4 +10,4 @@ flake8-import-order>=0.4.0, <0.6.0
 netifaces!=0.10.5
 nose
 pep8==1.5.7
-pep8-naming
+pep8-naming!=0.6.0
```
Comment (jstasiak) [inline requirements-dev.txt]: Can you split this into a separate pull request with an explanation as to why this is needed? Thanks.

File: .travis.yml (hunk 1 above)
Comment (mdiez-modus) [reply .travis.yml]: I needed this one to find out what version of flake you use so I could fix the build

Comment (mdiez-modus) [reply .travis.yml]: I need this so I could find out what version of flake you need so I could fix the flake errors

**Comment by mdiez-modus** [comment]
Hi!

As requested I split my PR into 2.

This one handles the exception on WSL and #131 makes flake8 happy again.

**Comment by jstasiak** [comment]
Thanks, merged.
